Sources/PostgresNIO/New/Connection State Machine/PreparedStatementStateMachine.swift

```diff
@@ -0,0 +1,68 @@
+import NIOCore
+
+struct PreparedStatementStateMachine {
+    enum State {
+        case preparing([PreparedStatementContext])
+        case prepared(RowDescription?)
+        case error(PSQLError)
```

Comment:
Here I keep track of `PSQLError` instead of a `PostgresBackendMessage.ErrorResponse` because that's what I get from the state machine that deals with statement preparation.